src/modules/nav/nav.component.ts

```diff
@@ -33,17 +33,20 @@ export class StacheNavComponent implements OnInit, StacheNav {
   }
 
   public isActive(route: any): boolean {
+    const activeUrl = this.router.url.split('#')[0];
     let path = route.path;
+    let navDepth: number;
 
     if (path.join) {
+      navDepth = path.length;
       path = path.join('/');
-    }
-
-    if (this.router.url.includes('#')) {
-      return (this.router.url.split('#')[0] === path);
     } else {
-      return (this.router.url === path);
+      navDepth = path.split('/').length;
     }
+
+    const isActiveParent = (navDepth > 2  && `${activeUrl}/`.indexOf(`${path}/`) === 0);
```

Comment:
On second look, the context for checking if a link is active is different, so a service wouldn't benefit us much just yet.